prov/shm/src/smr_msg.c

```diff
@@ -109,7 +109,7 @@ static ssize_t smr_generic_sendmsg(struct smr_ep *ep, const struct iovec *iov,
 	if (ret == -FI_ENOENT)
 		return -FI_EAGAIN;
 
-	ofi_spin_lock(&ep->tx_lock);
+	ofi_genlock_lock(&ep->util_ep.lock);
```

Comment:
I believe do_sar acquires the ep lock so you'll have to remove that to prevent from deadlocking

Reply:
this should have been caught in AWS CI... hmmm

Reply:
Was able to reproduce deadlock when I added `smr_domain->util_domain.threading = FI_THREAD_SAFE;` back in to smr_domain... it wasn't deadlocking b/c we weren't configuring threading in our tests.